src/compiler/checker.ts

```diff
@@ -1020,38 +1020,68 @@ namespace ts {
             return links.resolvedExports || (links.resolvedExports = getExportsForModule(moduleSymbol));
         }
 
-        function extendExportSymbols(target: SymbolTable, source: SymbolTable) {
+        interface ExportStarDiagnosticsLookupTable {
+            [id: string]: {specifierText: string, exportsWithDuplicate: ExportDeclaration[]};
+        }
+
+        function extendExportSymbols(target: SymbolTable, source: SymbolTable, lookupTable?: ExportStarDiagnosticsLookupTable, exportNode?: ExportDeclaration) {
             for (let id in source) {
                 if (id !== "default" && !hasProperty(target, id)) {
                     target[id] = source[id];
+                    if (lookupTable && exportNode) {
+                        lookupTable[id] = {
+                            specifierText: getTextOfNode(exportNode.moduleSpecifier),

```

Comment:
I'd prefer to use the`.text` because the message will look awkward if you use single quotes.